sql/updates/0.3.0--0.4.0.sql

```diff
@@ -1,3 +1,3 @@
-ALTER TABLE _timescaledb_catalog.dimension_slice
+ALTER TABLE IF EXISTS _timescaledb_catalog.dimension_slice
```

Comment:
So, the thing I wanted to avoid was to run through the entire lineage if we don't have to, which was the reason for this change to begin with. So, if a table is created and then deleted within a single extension update, we'd ideally never have to create it in the first place (unless we explicitly deem it necessary for correctness). 

Note that you still can create tables in modfiles, and you'd probably have to do it if you are, e.g., populating the data with info during the update (or, alternatively, we can add a "post" script when that need comes up).

But, in cases you are just adding new functionality that requires a new metadata table, you probably wouldn't have to do anything in modfiles.

In the end, I see clarity in knowing that everything that is installed via `CREATE EXTENSION` is also "installed" during updates, as opposed to excluding some stuff.

But if you really feel strongly about this I can exclude `schema.sql` and `tables.sql`.